File: PropEase-main/Project-main/db_connection.php
<?php
// Database connection details
$servername = "localhost"; // XAMPP runs on localhost
$username = "root";        // Default username for XAMPP
$password = "";            // Default password is empty
$dbname = "test";          // Your database name

// Create connection
try {
    $conn = new mysqli($servername, $username, $password, $dbname);

    // Check connection
    if ($conn->connect_error) {
        throw new Exception("Connection failed: " . $conn->connect_error);
    }

} catch (Exception $e) {
    die("Error: " . $e->getMessage());
}
?>

File: Propease_February2025/Project-main/db_connection.py
from flask import Flask
from flask_mysqldb import MySQL

# Initialize Flask app
app = Flask(__name__)

# Configure database connection details
app.config['MYSQL_HOST'] = 'localhost'         # Hostname
app.config['MYSQL_USER'] = 'root'             # Username
app.config['MYSQL_PASSWORD'] = ''             # Password
app.config['MYSQL_DB'] = 'propease'           # Database name
app.config['MYSQL_CURSORCLASS'] = 'DictCursor'  # Fetch data as dictionary for convenience

# Initialize MySQL object
mysql = MySQL(app)

# Test database connection
@app.route('/test-db-connection', methods=['GET'])
def test_db_connection():
    try:
        # Open a database cursor
        cursor = mysql.connection.cursor()
        cursor.execute("SELECT DATABASE();")
        db_name = cursor.fetchone()

        return {"status": "success", "message": f"Connected to database: {db_name['DATABASE()']}"}
    except Exception as e:
        return {"status": "error", "message": f"Database connection failed: {str(e)}"}

if __name__ == "__main__":
    app.run(debug=True)
